lib/src/main/java/io/ably/lib/transport/Hosts.java

```diff
@@ -40,39 +40,38 @@
      */
     public Hosts(String primaryHost, String defaultHost, ClientOptions options) throws AblyException {
         this.defaultHost = defaultHost;
-        if (primaryHost != null) {
+        this.fallbackHostsUseDefault = options.fallbackHostsUseDefault;
+        boolean hasCustomPrimaryHost = primaryHost != null && !primaryHost.equalsIgnoreCase(defaultHost);
+        String[] tempFallbackHosts = options.fallbackHosts;
+        if (options.fallbackHostsUseDefault) {
+            if (options.fallbackHosts != null) {
+                throw AblyException.fromErrorInfo(new ErrorInfo("fallbackHosts and fallbackHostsUseDefault cannot both be set", 40000, 400));
+            }
+            if (options.port != 0 && options.tlsPort != 0) {
```

Comment:
if (options.port != 0 || options.tlsPort != 0)

Reply:
Nice catch! The code was updated and one unit test was added.